crates/sui-framework/tests/delegation_tests.move

```diff
@@ -79,6 +83,133 @@ module sui::delegation_tests {
         test_scenario::end(scenario_val);
     }
 
+    #[test]
+    #[expected_failure(abort_code = validator_set::ENotAValidator)]
+    fun test_remove_delegation_post_active_flow_no_rewards() {
+        test_remove_delegation_post_active_flow(false)
+    }
+
+    #[test]
+    #[expected_failure(abort_code = validator_set::ENotAValidator)]
```

Comment:
`test_remove_delegation_post_active_flow` calls `request_add_delegation` twice and we aren't able to tell at which time this test fails with `ENotAValidator` code. Maybe add another test where we only add when active and withdraw when inactive, and don't do the second  `request_add_delegation`, so the test succeeds without aborts?

Reply:
Good point! Updated to add this test, and also simplified the other test with the handy helpers in `governance_test_utils`